src/opm/output/eclipse/Summary.cpp

```diff
@@ -1180,68 +1163,45 @@ Summary::Summary( const EclipseState& st,
     std::set< std::string > unsupported_keywords;
 
     for( const auto& node : sum ) {
-        const auto* keyword = smspec_node_get_keyword(node.get());
+        ecl_smspec_type * smspec = ecl_sum_get_smspec(this->ecl_sum.get());
+        std::string keyword = node.keyword();
 
-        const auto single_value_pair = single_values_units.find( keyword );
-        const auto funs_pair = funs.find( keyword );
-        const auto region_pair = region_units.find( keyword );
-        const auto block_pair = block_units.find( keyword );
+        const auto single_value_pair = single_values_units.find( keyword.c_str() );
+        const auto funs_pair = funs.find( keyword.c_str() );
+        const auto region_pair = region_units.find( keyword.c_str() );
+        const auto block_pair = block_units.find( keyword.c_str() );
```

Comment:
Why do you use `.c_str()` here?  These maps all use `std::string` as their key types.

Reply:
> Why do you use .c_str() here? These maps all use std::string as their key types.

I will update - I just tried to limit the amount of changes